tools/header_schema.py

```diff
@@ -2,38 +2,46 @@
 Generates the schema for an ORSO file,
 based on the Orso class (from orsopy.fileio.orso)
 """
+
+"""
+** NOTE **
+This script is extremely sensitive to the version of pydantic that is installed.
+It works with pydantic-1.10.13 and lower, but not with pydantic >= 2.0
```

Comment:
Yes, the bump to pydantic 2.0 is basically a complete rewrite of pydantic.  They broke a bunch of things that used to make dataclasses similar in functionality to pydantic BaseModel.  I think pydantic.dataclasses are not the top priority for them so they won't be brought back up to feature-parity with 1.x for a little while.

Reply:
I spoke too soon... looks like they have made everything work really well!  I hadn't tried it for a while.  All we have to do is
```python
from pydantic import TypeAdapter
...
    schema.update(TypeAdapter(Orso).json_schema())
```